qa/acceptance-tests/src/test/java/io/camunda/it/client/JobSearchTest.java

```diff
@@ -0,0 +1,659 @@
+
+    // then
+    assertThat(resultAsc.items())
+        .extracting(Job::getProcessInstanceKey)
+        .containsExactlyElementsOf(all.stream().sorted(Comparator.naturalOrder()).toList());
+    assertThat(resultDesc.items())
+        .extracting(Job::getProcessInstanceKey)
+        .containsExactlyElementsOf(all.stream().sorted(Comparator.reverseOrder()).toList());
+  }
+
+  @Test
+  void shouldSortJobsByElementId() {
+    // given
+    final var all =
+        camundaClient.newJobSearchRequest().send().join().items().stream()
+            .map(Job::getElementId)
+            .toList();
+    // when
+    final var resultAsc =
+        camundaClient.newJobSearchRequest().sort(s -> s.elementId().asc()).send().join();
+    final var resultDesc =
+        camundaClient.newJobSearchRequest().sort(s -> s.elementId().desc()).send().join();
+
+    // then
+    assertThat(resultAsc.items())
+        .extracting(Job::getElementId)
+        .containsExactlyElementsOf(all.stream().sorted(Comparator.naturalOrder()).toList());
+    assertThat(resultDesc.items())
+        .extracting(Job::getElementId)
+        .containsExactlyElementsOf(all.stream().sorted(Comparator.reverseOrder()).toList());
+  }
+
+  @Test
+  void shouldSortJobsByElementInstanceKey() {
+    // given
+    final var all =
+        camundaClient.newJobSearchRequest().send().join().items().stream()
+            .map(Job::getElementInstanceKey)
+            .toList();
+    // when
+    final var resultAsc =
+        camundaClient.newJobSearchRequest().sort(s -> s.elementInstanceKey().asc()).send().join();
+    final var resultDesc =
+        camundaClient.newJobSearchRequest().sort(s -> s.elementInstanceKey().desc()).send().join();
+
+    // then
+    assertThat(resultAsc.items())
+        .extracting(Job::getElementInstanceKey)
+        .containsExactlyElementsOf(all.stream().sorted(Comparator.naturalOrder()).toList());
+    assertThat(resultDesc.items())
+        .extracting(Job::getElementInstanceKey)
+        .containsExactlyElementsOf(all.stream().sorted(Comparator.reverseOrder()).toList());
+  }
+
+  @Test
+  void shouldSortJobsByTenantId() {
+    // given
+    final var all =
+        camundaClient.newJobSearchRequest().send().join().items().stream()
+            .map(Job::getTenantId)
+            .toList();
+    // when
+    final var resultAsc =
+        camundaClient.newJobSearchRequest().sort(s -> s.tenantId().asc()).send().join();
+    final var resultDesc =
+        camundaClient.newJobSearchRequest().sort(s -> s.tenantId().desc()).send().join();
+
+    // then
+    assertThat(resultAsc.items())
+        .extracting(Job::getTenantId)
+        .containsExactlyElementsOf(all.stream().sorted(Comparator.naturalOrder()).toList());
+    assertThat(resultDesc.items())
+        .extracting(Job::getTenantId)
+        .containsExactlyElementsOf(all.stream().sorted(Comparator.reverseOrder()).toList());
+  }
+
+  @Test
+  void shouldSearchByFromLimit() {
```

Comment:
❓ Does it make sense to sort them to avoid any potential flakiness due to different result orders in the database?